### PR TITLE
Add font-weight option to Mini Cart block text

### LINKED ISSUE #6675: Mini Cart button: allow changing font weight styles
Currently, the Mini Cart button is bold by default. We would like to allow users to customize the font weight styles.

## To reproduce

1. Create a post or page.
2. Add the Mini Cart block.
3. Notice the button text is bold and there is no way to make it regular.

## Expected behavior

I would expect to be able to change the text font weight.

### PR DESCRIPTION
**Pull request by danielwrobert**
Adds typography font-weight support to Mini Cart. Previously the rendered output would default to semi-bold (600) but this allows for that weight to be controlled from within the block options in the editor.

Fixes #6675 

### Screenshots

| Before | After |
| ------ | ----- |
|  CleanShot 2022-07-27 at 22 08 03    |   CleanShot 2022-07-29 at 12 15 15   |

#### Screen Recordings

Before

https://user-images.githubusercontent.com/481776/181405817-1f53803f-45bf-43a2-bab3-5cf8af16ae0a.mp4

After

https://user-images.githubusercontent.com/481776/181801095-8e88dd5b-17e0-4c1a-8243-4ada0a6e5a61.mp4

### Testing

1. Make sure you have a blocks theme active (like Twenty Twenty-Two).
2. Navigate to **Appearance > Editor (Beta)** and add a Mini Cart block somewhere in your site.
3. In the editor, select the Mini Cart block and, in the block settings, choose Font Weight under the Typography section.
4. Adjust the font weight to something other than the Default.
5. Confirm changes are present in the editor.
6. Save changes and preview the frontend to confirm the changes have persisted.

### WooCommerce Visibility

* [ ] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental

### Changelog

> Fix: Add font-weight controls to the Mini Cart block text.

### CODE REVIEW

File: assets/js/blocks/mini-cart/block.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ import QuantityBadge from './quantity-badge';
 import { MiniCartContentsBlock } from './mini-cart-contents/block';
 import './style.scss';
 import { blockName } from './mini-cart-contents/attributes';
+import { useTypographyProps } from '../../hooks/style-attributes';
```
Comment (sunyatasattva) [inline assets/js/blocks/mini-cart/block.tsx:37]: I know our repo is littered with those, but in our most recent meetup, we decided to try [avoiding relative paths as much as we can](https://teamkirigami.wordpress.com/2022/03/24/meetup-code-standards-and-practices-brainstorm/#relative-paths-vs-absolute-paths), unless the imports belong with each other.

Again, doing a quick search on exactly this import, I see there are many cases in which this is not respected, but I'd like to change that. Even more so as I kinda [took some responsibility for this](https://teamkirigami.wordpress.com/2022/03/24/meetup-code-standards-and-practices-brainstorm/#comment-248).

There is an alias `@woocommerce/base-hooks` that could be used. I see that this hook lives in a different directory as all our other hooks (which are under `js/base/hooks`). I think this might have been a mistake, coming from the fact that we are all new and we are still learning to navigate the codebase (but perhaps @gigitux can correct me if there was a specific reason to create another directory), especially likely given that this file was created quite some time ago when we were even newer.

So, perhaps, we could move this file into the `base/hooks` directory and change at least this import to use the alias?

What do you think?

File: assets/js/blocks/mini-cart/block.tsx [hunk 2]
```diff
@@ -44,14 +45,16 @@ interface Props {
...
-}: Props ): JSX.Element => {
+const MiniCartBlock = ( attributes: Props ): JSX.Element => {
+	const {
+		isInitiallyOpen = false,
+		colorClassNames,
+		style,
+		contents = '',
+		addToCartBehaviour = 'none',
+		hasHiddenPrice = false,
+	} = attributes;
```
Comment (sunyatasattva) [inline assets/js/blocks/mini-cart/block.tsx:56]: Just curious, why did you change the destructuring to be within the function body? To be clear, I also like it more, I wonder if there is a specific reason.

**Review by sunyatasattva — APPROVED** [review]
Thanks for the PR! Everything looks solid. I only left a comment regarding the imports, which I'd love to see fixed. But I don't want it to be a blocker if you feel otherwise, so I'm pre-approving. :)

File: assets/js/blocks/mini-cart/block.tsx (hunk 1 above)
Comment (danielwrobert) [reply assets/js/blocks/mini-cart/block.tsx:37]: > So, perhaps, we could move this file into the base/hooks directory and change at least this import to use the alias?

@sunyatasattva I'm in favor of this! I think that will add some consistency and the more we do that, the easier our codebase becomes to reason about overall.

That being said, because this relative path is used across a handful of other files that are unrelated to this PR, perhaps it would be better to circle back and target them all in one go in a separate PR. That way, we can focus and ensure there are no unintended side-effects or regressions from moving things around. Does that sound good?

I'm happy to handle that once I merge this in.

File: assets/js/blocks/mini-cart/block.tsx (hunk 2 above)
Comment (danielwrobert) [reply assets/js/blocks/mini-cart/block.tsx:56]: Good question!

It was adjusted so that we could access the whole `attributes` object directly for use in the `useTyporgraphyProps` hook[ here](https://github.com/woocommerce/woocommerce-blocks/pull/6760/files#diff-733abdc48b34d465d994c1bf4b179fdb74a9e061ae5bc020bd0f5db954a7b267R207).

File: assets/js/blocks/mini-cart/block.tsx (hunk 1 above)
Comment (danielwrobert) [reply assets/js/blocks/mini-cart/block.tsx:37]: I've added a new Issue to address this task at #6866.
